### PR TITLE
make qs_in a field called sediment__influx

### PR DESCRIPTION
**Pull request by gregtucker**
Make a field for previously private array _qs_in

Addresses #1369

### CODE REVIEW

File: landlab/components/erosion_deposition/generalized_erosion_deposition.py [hunk 1]
```diff
@@ -126,11 +134,9 @@ def __init__(
         self.initialize_output_fields()
 
         self._qs = grid.at_node["sediment__flux"]
+        self._qs_in = grid.at_node["sediment__influx"]
```
Comment (mcflugen) [inline landlab/components/erosion_deposition/generalized_erosion_deposition.py:137]: `self._qs_in` is unused. I recommend either exposing it as a property (with a descriptive name) or removing it.

**Review by mcflugen — CHANGES_REQUESTED** [review]

File: landlab/components/erosion_deposition/erosion_deposition.py [hunk 2]
```diff
@@ -130,6 +130,14 @@ class ErosionDeposition(_GeneralizedErosionDeposition):
             "doc": "Node array containing downstream-to-upstream ordered list of node IDs",
         },
         "sediment__flux": {
+            "dtype": float,
+            "intent": "out",
+            "optional": False,
+            "units": "m3/s",
+            "mapping": "node",
+            "doc": "Sediment flux (volume per unit time of sediment leaving each node)",
```
Comment (mcflugen) [inline landlab/components/erosion_deposition/erosion_deposition.py:138]: The description of the *sediment__flux* field is now inconsistent with other *landlab* components. The *LateralEroder* says that *sediment__flux* is, *"Sediment flux (volume per unit time of sediment __entering__ each node)"* but *Space* says that it is, *"Sediment flux (volume per unit time of sediment __leaving__ each node)"*.

I'm not sure which description is correct but we need to either update the descriptions or change some field names to make things consistent.

**Review by mcflugen — CHANGES_REQUESTED** [review]

File: landlab/components/erosion_deposition/generalized_erosion_deposition.py (hunk 1 above)
Comment (gregtucker) [reply landlab/components/erosion_deposition/generalized_erosion_deposition.py:137]: `self._qs_in` is defined here because it is used by derived classes.

Comment (mcflugen) [reply landlab/components/erosion_deposition/generalized_erosion_deposition.py:137]: In that case, how about making it public? That way, derived classes don't have to depend on the existence of a private variable. I think the safest way to do that would be through a property,
```python
@property
def sedflux_influx(self):
    return  self.grid.at_node["sediment__influx"]
```
Another option would be to drop the leading underscore and change the name to *qs_in* but then someone could accidentally change the pointer to the data.

File: landlab/components/erosion_deposition/erosion_deposition.py (hunk 2 above)
Comment (gregtucker) [reply landlab/components/erosion_deposition/erosion_deposition.py:138]: The new description is correct for ErosionDeposition and SPACE (both versions), but the old one is correct for LateralEroder! So we have a naming inconsistency... Here's a possible solution:
- change doc for sediment__flux to "Sediment flux (volume per unit time)"
- use sediment__influx if the flux is specified as coming into an element from adjacent elements
- use sediment__outflux if it is specified as leaving the element
- for backward compatibility, retain sediment__flux and have it refer to one of the above, whichever it did originally

Comment (mcflugen) [reply landlab/components/erosion_deposition/erosion_deposition.py:138]: It seems, to me anyway, the best option would be to use two different names depending on whether sediment in flowing into or out of grid cells. The pain will be to maintain backward compatibility.

What do you think?

File: landlab/components/erosion_deposition/generalized_erosion_deposition.py (hunk 1 above)
Comment (gregtucker) [reply landlab/components/erosion_deposition/generalized_erosion_deposition.py]: Agreed - changed as suggested

File: landlab/components/erosion_deposition/erosion_deposition.py (hunk 2 above)
Comment (gregtucker) [reply landlab/components/erosion_deposition/erosion_deposition.py]: Agreed. Names changed to sediment__influx and sediment__outflux. Pointer to one of these, as appropriate, with name sediment__flux is retained for backward compatibility until version 3.0.0+.

**Pull request merged**